zapgorm2.go

```diff
@@ -67,7 +71,7 @@ func (l Logger) Trace(ctx context.Context, begin time.Time, fc func() (string, i
 	}
 	elapsed := time.Since(begin)
 	switch {
-	case err != nil && l.LogLevel >= gormlogger.Error:
+	case err != nil && l.LogLevel >= gormlogger.Error && (!errors.Is(err, gorm.ErrRecordNotFound) || !l.IgnoreRecordNotFoundError):
```

Comment:
```suggestion
	case err != nil && l.LogLevel >= gormlogger.Error && (!l.IgnoreRecordNotFoundError || !errors.Is(err, gorm.ErrRecordNotFound)):
```

Small performance improvement by changing the check order

Reply:
done